codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -140,6 +143,100 @@ public void onPriorityTreeParentChanging(Http2Stream stream, Http2Stream newPare
                 }
             }
         });
+
+        if (listener == null) {
+            writeAllocatedBytesVisitor = new RemoteStreamVisitor() {
+                @Override
+                public boolean visit(Http2Stream stream) {
+                    state(stream).writeAllocatedBytes();
+                    return true;
+                }
+
+                @Override
+                public void beforeIteration() {
+                }
+
+                @Override
+                public Http2Stream streamMadeConnectionNotWritable() {
+                    return null;
+                }
+            };
+            incrementWindowSizeWritabilityVisitor = initialWindowSizeWritabilityVisitor =
+                    writePendingBytesWritabilityVisitor = null;
+        } else {
+            writeAllocatedBytesVisitor = new RemoteStreamVisitor() {
+                private boolean connectionOriginalWritable;
+                private Http2Stream streamMadeConnectionNotWritable;
+                @Override
+                public boolean visit(Http2Stream stream) {
+                    AbstractState state = state(stream);
+                    int written = state.writeAllocatedBytes();
```

Comment:
Can you remind me why we just write all allocated bytes without checking channel writability?